### PR TITLE
feat: scene metrics implementation overhaul + texture support

### PR DESCRIPTION
**Pull request by BrianAmadori**
## What does this PR change?

This PR introduces a SceneMetricsCounter overhaul that:

* Adds missing support for tracking textures as part of the metrics
* Finished new DataStore driven approach to counting that is lightweight and removes the need of having to search for the materials/textures/meshes every time something changes. Metrics for a particular asset are cached. Logic was simplified and code was removed as a result.
* Adds lots of new integration tests to ensure tracked scene metrics keeps behaving as expected. Now `SceneMetricsCounter` has unit tests too.

Also:
* Added small tool to measure worst metrics infractors over many scenes

Fixes:
* Fix: AssetPromiseKeeper not working right over many tests due to coroutine being stopped.
* Fix: PBRMaterial not cancelling texture promises if disposed before the promises were loaded
* Fix: materials being counted twice on some cases when Basic/PBRMaterial and ParametrizedShape are used together.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by AjimenezDCL — APPROVED** [review]
I left a couple of comments but I'm approving, good job!

File: unity-renderer/Assets/Scripts/MainScripts/DCL/DataStore/Variables/BaseHashSet.cs [hunk 1]
```diff
@@ -28,6 +28,14 @@ public void Add(T element)
...
     }
 
+    public void Add(IEnumerable<T> elements)
+    {
+        foreach ( T e in elements )
+        {
+            Add(e);
+        }
+    }
+
```
[the remark this answers is not in the document]
Comment (BrianAmadori) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/DataStore/Variables/BaseHashSet.cs:38]: Why? `Add` shouldn't be modifying the original HashSet enumerable, unless we do `Add( hashSet.Get() )` but that would be silly.

**Review by pravusjif — CHANGES_REQUESTED** [review]
Excellent job!

The only problem with this branch is something related to the global avatars scene that makes the world not render any avatar, it looks like an easy fix though
Screen Shot 2022-02-16 at 12 32 27

I think that error fucks up other things because the genesis plaza automatic-doors never open and I can't teleport anywhere, I'll try the build after the fix

**Review by pravusjif — APPROVED** [review]
Awsome! the issue has been solved!

**Pull request merged**